mega-zep-backend/src/main/java/com/gepardec/mega/service/impl/monthlyreport/MonthlyReportServiceImpl.java

```diff
@@ -48,14 +51,22 @@ public MonthlyReport getMonthendReportForUser(final String userId) {
         Employee employee = zepService.getEmployee(userId);
         final LocalDate date;
 
-        if (employee != null && employee.releaseDate() != null) {
-            date = LocalDate.parse(employee.releaseDate()).plusMonths(1);
+        if ((employee != null) && (employee.releaseDate() != null) && checkReleaseDate(employee.releaseDate())) {
+            date = LocalDate.parse(Objects.requireNonNull(employee.releaseDate())).plusMonths(1);
+
         } else {
             date = null;
         }
         return getMonthendReportForUser(userId, date);
     }
 
+    private boolean checkReleaseDate(String releaseDate) {
+        if (releaseDate.matches("^\\d\\d\\d\\d-{1}\\d\\d-{1}\\d\\d$")) {
```

Comment:
kommentar was das regex genau macht

return releaseDate.matches(..)

Reply:
@Ruhsi 
prüft ob das releaseDate String dem Datumsformat YYYY-mm-DD entspricht damit das parsen zu LocalDate reibungslos funktioniert. Das Fehlen dieses/eines Checks hat die Fehler in den MonthlyReportServiceImplTest verursacht.